packages/expo-updates/ios/EXUpdates/UpdatesUtils.swift

```diff
@@ -78,17 +52,171 @@ public final class UpdatesUtils: NSObject {
     return updatesDirectory
   }
 
-  internal static func sendEvent(toBridge bridge: RCTBridge?, withType eventType: String, body: [AnyHashable: Any]) {
-    guard let bridge = bridge else {
-      NSLog("EXUpdates: Could not emit %@ event. Did you set the bridge property on the controller singleton?", eventType)
-      return
+  /**
+   The implementation of checkForUpdateAsync().
+   The UpdatesService is passed in when this is called from JS through UpdatesModule
+   */
+  public static func checkForUpdate(_ block: @escaping ([String: Any]) -> Void) {
+    sendStateEvent(.check)
+    do {
+      let constants = try startAPICall()
+
+      var extraHeaders: [String: Any] = [:]
+      constants.database.databaseQueue.sync {
+        extraHeaders = FileDownloader.extraHeadersForRemoteUpdateRequest(
+          withDatabase: constants.database,
+          config: constants.config,
+          launchedUpdate: constants.launchedUpdate,
+          embeddedUpdate: constants.embeddedUpdate
+        )
+      }
+
+      let fileDownloader = FileDownloader(config: constants.config)
+      fileDownloader.downloadRemoteUpdate(
+        fromURL: constants.config.updateUrl!,
+        withDatabase: constants.database,
+        extraHeaders: extraHeaders
+      ) { updateResponse in
+        if let updateDirective = updateResponse.directiveUpdateResponsePart?.updateDirective {
+          switch updateDirective {
+          case is NoUpdateAvailableUpdateDirective:
+            block([:])
+            sendStateEvent(.checkCompleteUnavailable)
+            return
+          case is RollBackToEmbeddedUpdateDirective:
+            let body = [
+              "isRollBackToEmbedded": true
+            ]
+            block(body)
+            sendStateEvent(.checkCompleteAvailable, body: body)
+            return
+          default:
+            return handleCheckError(UpdatesUnsupportedDirectiveException(), block: block)
```

Comment:
we should throw this exception so that it uses the default error handling provided by `AsyncFunction` when an exception of this type is thrown (one that extends expo-modules-core Exception type). ditto with `handleFetchError`.